tests/integration/models/subscriptions.rs

```diff
@@ -0,0 +1,100 @@
+use crate::utils::{random_ascii_string, random_currency_or_country, random_price, spawn_app};
+use lib::models::subscriptions::{Subscription, SubscriptionModel};
+use pretty_assertions::assert_eq;
+use serde_json::json;
+use time::{Duration, OffsetDateTime};
+use uuid::Uuid;
+
+fn make_fake_sub() -> Subscription {
+    Subscription {
+        id: Uuid::new_v4(),
+        flow_id: random_ascii_string(),
+        subscription_id: random_ascii_string(),
+        report_timestamp: OffsetDateTime::now_utc(),
+        subscription_created: OffsetDateTime::now_utc() - Duration::hours(35),
+        fxa_uid: random_ascii_string(),
+        quantity: 1,
+        plan_id: random_ascii_string(),
+        plan_currency: random_currency_or_country(),
+        plan_amount: random_price(),
+        country: Some(random_currency_or_country()),
+        aic_id: Some(Uuid::new_v4()),
+        aic_expires: Some(OffsetDateTime::now_utc()),
+        cj_event_value: Some(random_ascii_string()),
+        status: Some(random_ascii_string()),
+        status_history: Some(json!([])),
+    }
+}
+
+#[tokio::test]
+async fn test_subscription_model_create_from_subscription_struct() {
+    let app = spawn_app().await;
+    let model = SubscriptionModel {
+        db_pool: &app.db_connection(),
+    };
+    let sub = make_fake_sub();
+    model
+        .create_from_sub(&sub)
+        .await
+        .expect("Failed to create test object.");
+    let result = model
+        .fetch_one_by_id(&sub.id)
+        .await
+        .expect("Could not fetch from DB.");
+    assert_eq!(result, sub);
+}
+
+#[tokio::test]
+async fn test_subscription_model_fetch_ones_by_unique_ids() {
+    let app = spawn_app().await;
+    let model = SubscriptionModel {
+        db_pool: &app.db_connection(),
+    };
+    let sub = make_fake_sub();
+    model
+        .create_from_sub(&sub)
+        .await
+        .expect("Failed to create test object.");
+    let result = model
+        .fetch_one_by_id(&sub.id)
+        .await
+        .expect("Could not fetch from DB.");
+    assert_eq!(sub, result);
+    let result = model
+        .fetch_one_by_flow_id(&sub.flow_id)
+        .await
+        .expect("Could not fetch from DB.");
+    assert_eq!(sub, result);
+}
+
+#[tokio::test]
+async fn test_subscription_model_fetch_ones_if_not_available() {
+    let app = spawn_app().await;
+    let model = SubscriptionModel {
+        db_pool: &app.db_connection(),
+    };
+    let sub = make_fake_sub();
+    model
+        .create_from_sub(&sub)
+        .await
+        .expect("Failed to create test object.");
+    let bad_id = Uuid::new_v4();
+    let result = model.fetch_one_by_id(&bad_id).await;
+    match result {
```

Comment:
Is it not possible to just assert based on the type of the error instead of matching?

Reply:
I have looked into this again and concluded not-cleanly-not-easily - in part depends on the traits implemented by the sqlx error.